recipes/glsl-optimizer/all/conanfile.py

```diff
@@ -0,0 +1,57 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+
+class GLSLOptimizerConan(ConanFile):
+    name = "glsl-optimizer"
+    description = "GLSL optimizer based on Mesa's GLSL compiler. Used in Unity for mobile shader optimization."
+    homepage = "https://github.com/aras-p/glsl-optimizer"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "glsl", "opengl")
+    license = "MIT"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt"
+
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            self.requires("getopt-for-visual-studio/20200201")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob(self.name + "-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
```

Comment:
force it only when required

Reply:
@uilianries 
I'm reading the cmake documentation again, and it seems that it is only effective when building shared libraries.
See https://cmake.org/cmake/help/latest/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html

What it does is collect all symbols of the object files, and pass it to the linker to export these symbols.
But when creating a static library, no linking is done. So the symbols are not tagged as exported.

Reply:
Okay, thanks for looking at this.